src/index.js

```diff
@@ -1,8 +1,59 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import promise from 'redux-promise';
+import reduxThunk from 'redux-thunk';
+import { Provider } from 'react-redux';
+import { createStore, applyMiddleware } from 'redux';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import jwt from 'jsonwebtoken';
+import logger from 'redux-logger'
+
 import './index.css';
-import App from './App';
 import registerServiceWorker from './registerServiceWorker';
+import reducers from './reducers';
+
+import NavbarComponent from './components/universal/navbar';
+import Homepage from './components/homepage';
+import Signin from './components/auth/signin';
+import Signup from './components/auth/signup';
+import Dashboard from './components/universal/dashboard';
+import requireAuth from './components/auth/require_auth';
+import noRequireAuth from './components/auth/no_require_auth';
+import { AUTHENTICATED } from './actions';
+
+const createStoreWithMiddleware = applyMiddleware(promise, reduxThunk,logger)(createStore);
+const store = createStoreWithMiddleware(
+    reducers, 
+    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
+);
```

Comment:
It's often best to move this logic

```
const createStoreWithMiddleware = applyMiddleware(promise, reduxThunk,logger)(createStore);
const store = createStoreWithMiddleware(
reducers, 
window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
 +);
```
and the reducers/index.js logic into a `store.js` file